x/leverage/keeper/collateral.go

```diff
@@ -78,6 +78,33 @@ func (k Keeper) CalculateCollateralValue(ctx sdk.Context, collateral sdk.Coins)
 	return total, nil
 }
 
+// VisibleCollateralValue uses the price oracle to determine the value (in USD) provided by
+// collateral sdk.Coins, using each token's uToken exchange rate. Always uses spot price.
+// Unlike CalculateCollateralValue, this function will not return an error if value calculation
+// fails on a token - instead, that token will contribute zero value to the total.
+func (k Keeper) VisibleCollateralValue(ctx sdk.Context, collateral sdk.Coins) sdk.Dec {
+	total := sdk.ZeroDec()
+
+	for _, coin := range collateral {
+		// convert uToken collateral to base assets
+		baseAsset, err := k.ExchangeUToken(ctx, coin)
+		if err != nil {
+			continue
+		}
+
+		// get USD value of base assets
+		v, err := k.TokenValue(ctx, baseAsset, types.PriceModeSpot)
+		if err != nil {
+			continue
```

Comment:
let's log this error with an appropriate message.

Reply:
Added logging at info level so it doesn't flood the error level with noise